File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int solution(string s)
{
    int z = 0;
    stack<char> stk;
    for (const auto& c: s)
    {
        if (!stk.empty())
        {
            if (stk.top() == c) stk.pop();
            else stk.push(c);
        }
        else stk.push(c);
    }
    return stk.empty() ? 1 : 0;
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution{
    public int solution(String s){
        Deque<Character> stk = new ArrayDeque<>();
        
        for (char c : s.toCharArray()) {
            if (stk.isEmpty()) {
                stk.push(c);
            } else if (stk.peek() == c) {
                stk.pop();
            } else {
                stk.push(c);
            }
        }
        
        return stk.isEmpty() ? 1 : 0;
    }
}